File: Clase1/src/clase1/Lista_simple.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package clase1;

/**
 *
 * @author Alex Rose
 */
public class Lista_simple {

    Nodo_LS inicio;

    public Lista_simple() {
        this.inicio = null;
    }

    //Metodo para insertar valores al final///
    public void InsertarFinal(String valor) {
        Nodo_LS nuevo = new Nodo_LS(valor);
        if (inicio == null) {//Si la lista se encuentra vacia
            inicio = nuevo;
        } else {//si la lista no esta vacia
            Nodo_LS auxActual = inicio;

            while (auxActual != null) {
                if (auxActual.Siguiente == null) {
                    auxActual.Siguiente = nuevo;
                    break;
                }
                auxActual = auxActual.Siguiente;
            }
        }
    }
    
    
    //Metodo para insertar valores al final///
    public void InsertarOrden(String valor) {
        Nodo_LS nuevo = new Nodo_LS(valor);
        if (inicio == null) {//Si la lista se encuentra vacia
            inicio = nuevo;
        } else {//si la lista no esta vacia
            Nodo_LS auxActual = inicio;
            Nodo_LS auxSiguiente;
            while (auxActual != null) {
              auxSiguiente=auxActual.Siguiente;
              if(nuevo.valor.compareTo(auxActual.valor)<0){//insertar al inicio de la lista por que es menor
                  nuevo.Siguiente=auxActual;
                  inicio=nuevo;
                  break;
              }else if(auxSiguiente==null){//insertar al final de la lista
                  auxActual.Siguiente=nuevo;
                  break;
              }else if(nuevo.valor.compareTo(auxSiguiente.valor)<0){//insertar en medio de la lista
                  auxActual.Siguiente=nuevo;
                  nuevo.Siguiente=auxSiguiente;
                  break;
              }
              
              auxActual=auxActual.Siguiente;
              
            }
        }
    }

    //meto para imprimir la lista
    public void Imprimir() {
        Nodo_LS auxActual = inicio;

        while (auxActual != null) {
            System.out.print("["+auxActual.valor+"]->");
            auxActual = auxActual.Siguiente;
        }
        
        System.out.println("NULL");

    }

}

File: Clase3/ListaSimple.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   ListaSimple.cpp
 * Author: Alex Rose
 * 
 * Created on 6 de agosto de 2022, 09:42 AM
 */
#include "ListaSimple.h"

void ListaSimple::InsertarFinal(int valor) {
    nodosimple*nuevo = new nodosimple();
    nuevo->valor = valor;

    if (Inicio == NULL) {
        Inicio = nuevo;
    } else {
        nodosimple*auxActual = Inicio;

        while (auxActual != NULL) {
            if (auxActual->sig == NULL) {
                auxActual->sig = nuevo;
                break;
            }
            auxActual = auxActual->sig;
        }
    }


}

void ListaSimple::InsertarEnOrden(int valor) {
    nodosimple*nuevo = new nodosimple();
    nuevo->valor = valor;
    if (Inicio == NULL) {//Si la lista se encuentra vacia
        Inicio = nuevo;
    } else {//si la lista no esta vacia
        nodosimple*auxActual = Inicio;
        nodosimple*auxSiguiente;
        while (auxActual != NULL) {
            auxSiguiente = auxActual->sig;
            if (nuevo->valor < auxActual->valor) {//insertar al inicio de la lista por que es menor
                nuevo->sig = auxActual;
                Inicio = nuevo;
                break;
            } else if (auxSiguiente == NULL) {//insertar al final de la lista
                auxActual->sig = nuevo;
                break;
            } else if (nuevo->valor < auxSiguiente->valor) {//insertar en medio de la lista
                auxActual->sig = nuevo;
                nuevo->sig = auxSiguiente;
                break;
            }
            auxActual = auxActual->sig;
        }
    }
}

void ListaSimple::Imprimir() {
    nodosimple*aux = Inicio;
    while (aux != NULL) {
        cout <<"[" << aux->valor << "]->";
        aux = aux->sig;
    }
    cout << ("NULL");
}

